### PR TITLE
feat: luarocks support

### PR DESCRIPTION
**Pull request by mrcjkb**
Hey :wave: 

### Summary

This PR is part of a push to get neovim plugins on [LuaRocks](https://luarocks.org/labels/neovim).

See also:

- [rocks.nvim](https://github.com/nvim-neorocks/rocks.nvim), a new luarocks-based plugin manager.
- [this blog post](https://mrcjkb.github.io/posts/2023-01-10-luarocks-tag-release.html).

### Things done:

* Add a release-please workflow that creates release PRs with SemVer versioning based on [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).
* Add a workflow that publishes tags to LuaRocks when a tag or release is pushed.

The workflows are based on [this guide](https://github.com/vhyrro/sample-luarocks-plugin) by @vhyrro.

### Notes:

* On each merge to `main`, the `release-please` workflow creates (or updates an existing) release PR.
* You decide when to merge release PRs.
  Doing so will result in a SemVer tag, and a GitHub release, which will trigger the `luarocks` workflow.
* In the `luarocks` github action, tagged releases are installed locally and then published to luarocks.org.
* For the luarocks workflow to work, someone with a luarocks.org account will have to add their [API key](https://luarocks.org/settings/api-keys) to this repo's [GitHub actions secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository).
* Due to a shortcoming in LuaRocks (https://github.com/luarocks/luarocks-site/issues/188), the `neovim` and/or `vim` labels have to be added  to the LuaRocks package manually (after the first upload), for this plugin to show up in https://luarocks.org/labels/neovim or https://luarocks.org/labels/vim, respectively.

### CODE REVIEW

**Comment by mrcjkb** [comment]
I ran prettier, but CI is still failing.

**Comment by mochaaP** [comment]
@mrcjkb made some changes, please review :)

File: .github/workflows/luarocks.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+jobs:
+  luarocks-upload:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Get Version
+        run: |
+          if ! git describe --exact-match --tags; then
+            echo "LUAROCKS_VERSION=$(date +v%Y.%m.%d)" >> "${GITHUB_ENV}"
```
Comment (mrcjkb) [inline .github/workflows/luarocks.yml:16]: :thinking: I'm not sure if this is a good idea for luarocks, which expects SemVer.
If you release version `1.0.0` one day, then `luarocks install none-ls.nvim` will see `2024.4.2` as a greater version than the actual tagged release.

I would suggest to just publish an `scm` rockspec if you don't want to commit to SemVer tags yet.

```suggestion
            echo "LUAROCKS_VERSION='scm'" >> "${GITHUB_ENV}"
```

You could also simplify this by setting

```yaml
with:
    version: "scm"
```

and omitting the `LUAROCKS_VERSION` for now.

Comment (mochaaP) [reply .github/workflows/luarocks.yml]: done in 1230cf2b46a7a8fada0d129433f7b456368a7d3a

**Comment by mochaaP** [comment]
btw, would be nice if you could also add this to none-ls-extras

**Comment by mochaaP** [comment]
<https://github.com/nvimtools/none-ls.nvim/actions/runs/8531392590/job/23370952901>

**Comment by mrcjkb** [comment]
> https://github.com/nvimtools/none-ls.nvim/actions/runs/8531392590/job/23370952901 scm publishing is broken @mrcjkb

https://github.com/nvimtools/none-ls.nvim/pull/106

**Pull request merged**